pkg/generate/code/set_resource.go

```diff
@@ -1229,6 +1238,29 @@ func SetResourceForStruct(
 	for memberIndex, memberName := range sourceShape.MemberNames() {
```

Comment:
I was expecting you to change the above to loop over the *targetShape*'s members, not the source shape...

Reply:
I know we originally discussed that approach, but the implementation was more convoluted so I pivoted to what I thought would be more straightforward given the current codebase. 

I know we'll have a *code-generator* refactor task of iterating over resource fields instead of shape members in `SetResource` -- would you mind if I made this update to iterate over _targetShape_ members at that time? I think those patterns are similar and consistent and be better understood if delivered around the same time.